### PR TITLE
Add Fake IdP SAML App tokens: Frontend

### PR DESCRIPTION
**Pull request by wleightond**
## Proposed changes
This PR adds the frontend components for fake IdP SAML2 app tokens. 

These are installed as a SAML app on the user's Identity Provider dashboard. When an attacker clicks on the app a SAML request is sent to the canarytokens server with the user's identity, and an alert is triggered.

Changes:
- Add IdP token creation / management frontend UI

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Checklist
- [x] Lint and unit tests pass locally with my changes (if applicable)
- [x] I have run pre-commit (`pre-commit` in the repo)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation in [canarytokens-docs PR 57](https://github.com/thinkst/canarytokens-docs/pull/57)

### CODE REVIEW

File: frontend_vue/src/components/tokens/idp_app/howToUse.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export const howToUse = [
+  'Create a new SAML2 app in your Identity Provider (e.g. Okta).',
+  'Create a new token and use the login URL and entity ID provided.',
+  'Give it the name of a common service you don\'t use (e.g. Salesforce).',
+  'Optionally, upload a realistic icon for the service.',
```
Comment (mclmax) [inline frontend_vue/src/components/tokens/idp_app/howToUse.ts:5]: there are each meant to be unique ideas not paragraphs

File: frontend_vue/src/components/tokens/idp_app/CreateAppPreview.vue [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    value: 'Zendesk',
+    label: 'idp_icons/zendesk.png'
+  },
+  {
+    value: 'Freshbooks',
+    label: 'idp_icons/freshbooks.png'
+  },
+  {
+    value: 'Elasticsearch',
+    label: 'idp_icons/elasticsearch.png'
```
Comment (mclmax) [inline frontend_vue/src/components/tokens/idp_app/CreateAppPreview.vue:150]: this icon doesn't git

File: frontend_vue/src/components/tokens/idp_app/CreateAppPreview.vue [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    value: 'GitHub',
+    label: 'idp_icons/github.png'
+  },
+  {
+    value: 'Slack',
+    label: 'idp_icons/slack.png'
+  },
+  {
+    value: 'MS Teams',
+    label: 'idp_icons/ms-teams.png'
```
Comment (mclmax) [inline frontend_vue/src/components/tokens/idp_app/CreateAppPreview.vue:206]: this isn't a png and has the grid background included

File: frontend_vue/src/components/base/BaseDownloadIconButtom.vue [hunk 4]
```diff
@@ -0,0 +1,35 @@
+<template>
```
Comment (mclmax) [inline frontend_vue/src/components/base/BaseDownloadIconButtom.vue:1]: this needs an animation something similar to the copy button

**Review by mclmax — CHANGES_REQUESTED** [review]

File: frontend_vue/src/main.ts [hunk 5]
```diff
@@ -1,10 +1,7 @@
 import { createApp } from 'vue';
 import App from './App.vue';
 import router from './router';
-import {
-  library,
-  type IconDefinition,
```
[the remark this answers is not in the document]
Comment (wleightond) [reply frontend_vue/src/main.ts:6]: Not as far as I can see.

File: frontend_vue/src/utils/tokenServices.ts [hunk 6]
```diff
@@ -420,4 +420,20 @@ export const tokenServices: TokenServicesType = {
     ],
     keywords: ['microsoft', 'css', 'cloned']
   },
+  [TOKENS_TYPE.IDP_APP]: {
+    label: 'SAML IdP App',
+    description:
+      'Get an alert when an attacker opens a fake app from your Identity Provider dashboard.',
+    documentationLink: '',
```
[the remark this answers is not in the document]
Comment (wleightond) [reply frontend_vue/src/utils/tokenServices.ts]: Yes! Thanks :) turns out the one I copied from had it missing too. Added to both.

File: frontend_vue/src/components/tokens/idp_app/ActivatedToken.vue [hunk 7]
```diff
@@ -0,0 +1,30 @@
+<template>
+  <div v-if="!tokenUrl || !entityId">Error loading</div>
```
[the remark this answers is not in the document]
Comment (wleightond) [reply frontend_vue/src/components/tokens/idp_app/ActivatedToken.vue]: [ditto](https://github.com/thinkst/canarytokens/pull/623#discussion_r1867745913)

File: frontend_vue/src/components/base/BaseLabelArrow.vue [hunk 8]
```diff
@@ -1,6 +1,7 @@
 <!-- eslint-disable vue/no-v-html -->
 <template>
   <BaseLabel
+    v-bind="$attrs"
```
[the remark this answers is not in the document]
Comment (vittoriaThinkst) [reply frontend_vue/src/components/base/BaseLabelArrow.vue:4]: This seems to work fine everywhere, tests are passing too

File: frontend_vue/src/components/tokens/idp_app/ActivatedToken.vue (hunk 7 above)
[the remark this answers is not in the document]
Comment (wleightond) [reply frontend_vue/src/components/tokens/idp_app/ActivatedToken.vue]: Found a way to remove.

File: frontend_vue/src/components/tokens/idp_app/ManageToken.vue [hunk 9]
```diff
@@ -0,0 +1,23 @@
+<template>
+  <div v-if="!tokenUrl || !entityId">Error loading</div>
```
[the remark this answers is not in the document]
Comment (wleightond) [reply frontend_vue/src/components/tokens/idp_app/ManageToken.vue]: Found a way to remove.

File: frontend_vue/src/components/base/BaseCopyButton.vue [hunk 10]
```diff
@@ -5,8 +5,14 @@
       shown: isTriggered,
       triggers: tooltipTriggers,
     }"
-    class="h-[2rem] w-[2rem] font-semibold text-white rounded-full bg-green hover:bg-green-300 transition duration-100"
+    :class="
+      props.disabled
+        ? 'cursor-not-allowed bg-grey-200 hover:bg-grey-200'
+        : 'cursor-pointer  bg-green hover:bg-green-300 '
+    "
+    class="h-[2rem] w-[2rem] font-semibold text-white rounded-full transition duration-100"
```
[the remark this answers is not in the document]
Comment (wleightond) [reply frontend_vue/src/components/base/BaseCopyButton.vue]: no longer needed since the icon is now selected at generation time and will always be present

**Pull request merged**